### ISSUE TITLE
4.0 PRESENTATION: hard to see separation between action buttons and data

### ISSUE THREAD

**Issue by sunnysideup** [silverstripe/silverstripe-framework#7122]
In the model admin, it is hard to see the separation between the action buttons at the top and the actual data grid (4.0 beta 1).

I am wondering if a small line, subtle between these sections would make this better.

**Comment by chillu** [comment]
@clarkepaul Do you agree? @sunnysideup When filing UX issues, please always include a screenshot

**Comment by sunnysideup** [comment]
here are some ideas:
![image](https://user-images.githubusercontent.com/167154/27843523-b62454aa-6168-11e7-8b62-2e753a932d7a.png)

ROW 1: choose model / navigate

ROW 2: work with current data-set including search

ROW 3: the actual data set

Here is the original with a subtle line:
![image](https://user-images.githubusercontent.com/167154/27843559-2955574e-6169-11e7-96ed-d21f4a3a88d4.png)


and without:
![image](https://user-images.githubusercontent.com/167154/27843563-345507ac-6169-11e7-91c9-99ce3bf202ea.png)

**Comment by clarkepaul** [comment]
Unfortunately, I disagree. We can have multiple gridfields on a page and the actions need to more visually connect to the gridfield than to the overall page. If they are connected more to the top bar it feels like the actions are global actions for that section eg. a page might contain a gridfield nested amongst other things.

I will be monitoring how users respond to using the actions to see if things are not clear but from what I can tell (so far) there hasn't been any actual concerns from real users. I have had some concerns from developers thinking users won't understand with similar things like buttons not looking like buttons. The fact is that traditional patterns for UI's have evolved and open looking UI's are much more common.

My reasoning for simplifying the UI look is to declutter and minimise separations when they are not needed. This gives the CMS more of a simplified appearance even though the complexity still exists (hopefully without the affordance of removing usability).  This is still early days for the UI so I will keep it in mind. FYI I think the gridfield still needs work (when and how titles are used and their connection to the actions so by no means is the UI finished).

**Comment by chillu** [comment]
Agree with Paul, shading the actions area is not the right way to go due to multiple GridFields. I do think there's too much space between the actions and the table to visually connect these as one logical unit. Was there a reason for this, Paul?

![image](https://user-images.githubusercontent.com/111025/27846268-f9e177ce-6189-11e7-84e8-095806bad03e.png)

**Comment by sunnysideup** [comment]
@clarkepaul

I did not know you can have two gridfield tables on one modelAdmin.  That is new to me.  It seems like and edge case to me. However, assuming that is important than I totally agree with all comments above However, there are two things that still need attention from my perspective

* As Ingo said, the space between the buttons and the table is too much for them to be logically connected.  They should be closer and ideally the table actions buttons (print, etc...) should be further away from the top select model row.

* it seems to me that the search button should be on the same line as export / import / add or below.  Its current position is confusing. The closer it is to the data-set the better.  What is more, if we were to have more than one gridfield then the search button is definitely in a strange place. 

Keep in mind that I have just started to look at 4.0.  Obviously I may have missed a lot of important stuff, but at the same time it allows me to have a fresh look.  

My first impression of the ModelAdmin was: there are a lot of buttons / links here, but it is hard to see how they relate and/or what is what.  The main this is to separate the "select model row" (top row) from the model current being edited.  I tried to do this with the Dark Blue example.

**Comment by chillu** [comment]
> I did not know you can have two gridfield tables on one modelAdmin

You can't, but we're trying to create consistent UIs, not something specific to ModelAdmin - this is a GridField which just happens to display search results as the only form element in a ModelAdmin.

**Comment by sunnysideup** [comment]
> You can't, but we're trying to create consistent UIs, not something specific to ModelAdmin - this is a GridField which just happens to display search results as the only form element in a ModelAdmin.

So I'd say:

* GridField actions button closer to GridField itself
* make ModelAdmin Search an integral part of a GridField and not the ModelAdmin Model Selector (not sure if that would work for 4.0 since we are in Beta now).

**Comment by clarkepaul** [comment]
Yeah the spacing is an issue, I only gave it a quick style and didn't get into the scenarios of when Gridfields need a heading below the actions and when it is in the Toolbar. We did a temporary fix to hide the heading between actions and Gridfield but I think the cell still exists which is holding the space (sometimes there is the text "viewing 1-15 or 25") top right of grid as well which needs to be considered.

**Comment by sunnysideup** [comment]
Ok - I reckon that should be improved, because it is such a key part of the CMS.

What about the search button?

**Comment by chillu** [comment]
@sunnysideup Wanna send a PR for that spacing issue?

**Comment by clarkepaul** [comment]
@sunnysideup I'm a bit confused about the position of the search you are referring to. In gridfields search is currently in the table header (not in the row of actions) and that is good from what I can tell. In Pages and Files sections it is in the top right corner, but I take it you think it should be in the row of actions? Perhaps a mockup might make it clearer.

**Comment by sunnysideup** [comment]
I am talking about the ModelAdmin search:
https://user-images.githubusercontent.com/167154/27843563-345507ac-6169-11e7-91c9-99ce3bf202ea.png
In this situation, the search would make a lot more sense in the action button row - but table header maybe even better. 

Here is an example of a search in the action button row:
https://user-images.githubusercontent.com/167154/27843523-b62454aa-6168-11e7-8b62-2e753a932d7a.png

**Comment by chillu** [comment]
I'd like to keep the special case styles to a minimum (GridField search bar in ModelAdmin). Adding a grey background would be fine (doesn't increase testing variations), but: Moving the search into the GridField is difficult because it's actually part of the ModelAdmin UI, not the GridField itself. The proper way to do this would be a new GridFIeld component for "expandable search", which can be used as an alternative to the inlined column title search. That'd be awesome actually. But just moving ModelAdmin UI into GridField markup somehow will make the system more hacky.

**Comment by sunnysideup** [comment]
@chillu - of course we should not have a hacky solution, but we need to put users first, not developer sensibilities.  I reckon the search in the ModelAdmin is in a weird place and from a user's perspective it is also an exception (search is usually with gridfield).
